x-pack/plugins/security_solution/server/lib/detection_engine/rule_monitoring/logic/rule_execution_log/create_rule_execution_summary.ts

```diff
@@ -13,41 +13,51 @@ import {
   ruleExecutionStatusToNumber,
   RuleExecutionStatus,
 } from '../../../../../../common/detection_engine/rule_monitoring';
+
 import type { RuleParams } from '../../../rule_schema';
 
 export const createRuleExecutionSummary = (
   rule: SanitizedRule<RuleParams> | ResolvedSanitizedRule<RuleParams>
 ): RuleExecutionSummary | null => {
+  if (!rule.monitoring) {
+    // In the rule type the monitoring object is optional because in some cases rules client returns a rule without it.
+    // For instance, when we call RulesClient.create(). Despite the fact that it's always in the rule saved object,
+    // even when it was just created:
+    // https://github.com/elastic/kibana/blob/26eddec70844ffb65e5d3521a7adb52e643c4534/x-pack/plugins/alerting/server/rules_client/methods/create.ts#L155
+    return null;
+  }
+
+  const lastRunInternal = rule.monitoring.run.last_run;
```

Comment:
Basically it makes a bit confusing regarding the difference of `lastRunInternal` and `lastRunPublic`. As far as I see we use `timestamp` and some metrics from `lastRunInternal` so we could improve readability by making it straightforward right here

```
const { timestamp: lastRunTimestamp, metrics } = rule.monitoring.run.last_run;

...

date: lastRunTimestamp,
...
total_indexing_duration_ms: metrics.total_indexing_duration_ms ?? undefined,
total_search_duration_ms: metrics.total_search_duration_ms ?? undefined,
execution_gap_duration_s: metrics.gap_duration_s ?? undefined,
...
```

Reply:
@maximpn I don't think this would make it less confusing. It's still two different objects within the rule object and they have the same name for some reason... I added some comments clarifying the differences.